### ISSUE TITLE
replicating results

### ISSUE THREAD

**Issue by oarriaga** [MarkPKCollier/NeuralTuringMachine#5]
Hi!

I have been trying to replicate the copy-task results from your [paper](https://arxiv.org/abs/1807.08518); however, after running `python3 run_tasks.py --experiment_name=trial01` the copy task does not seem to converge.

I obtained an error of 31.89375 per sequence and it gave the following results:
![trial01_20_0](https://user-images.githubusercontent.com/5577137/52638726-e0bcf300-2ed2-11e9-8dd4-53083fe07db1.png)

I ran it now multiple times but I always got similar results (I believe that you ran the experiments 10 times and took the mean per plot).

I checked the defaults parameters but everything seems to be in accordance to your paper.

Is there something that I am missing?

By the way, great job with the project :-)!

**Comment by MarkPKCollier** [comment]
Hi,

Thanks for getting in touch.

In the paper we used the AdamOptimizer with a learning rate of 0.001 - are you using the default RMSProp optimizer (sorry probably should change Adam to the default)?

BTW the plots in the paper are the elementwise median of 10 runs. There is still some instability in training with the improved memory initialization scheme but the "constant" init scheme is much more stable than the alternatives.

Let me know if using Adam fixes it, if not feel free to send on some more details on the TF version, etc.

**Comment by xiaoshingshing** [comment]
I used Adam and got exactly the same problem.
I use the default settings in the code, except that I change the 'num_train_steps' 10 times larger.
The error converge to about 24 per sequence and is far beyond 0.

**Comment by oarriaga** [comment]
I found that I was not setting the flag --man properly to 'ntm'. I believe it would be a good a idea to have 'ntm' as default rather than 'none'.

**Comment by MarkPKCollier** [comment]
Yes that should have been the default, changed now thanks.
